gpytorch/lazy/cat_lazy_tensor.py

```diff
@@ -185,7 +185,18 @@ def _get_indices(self, row_index, col_index, *batch_indices):
 
     def _getitem(self, row_index, col_index, *batch_indices):
         indices = [*batch_indices, row_index, col_index]
-        cat_dim_indices = indices[self.cat_dim]
+        cat_dim = self.cat_dim % len(self.shape)
```

Comment:
Is this something we'd want to do at a higher level (e.g. upon init)?

Reply:
`cat_dim` is actually normalized to a negative index in the init, and here I'm making it positive again for convenience.

When I originally looked at this code, I thought the reason `cat_dim` is stored as a negative index is to avoid having to do adjustments to it when when indexing through the cat... but on second thought, adjustments would still be needed, e.g.:

```
M = zeros(size = (2, 3, 4, 5, 6))
C = cat([M, M], dim = -4)
C[:,:,0,:,:] == cat([M[:,:,0,:,:], M[:,:,0,:,:]], dim = -3) # -4 -> -3
```

(I haven't had time to take a deeper look since making this PR.)